pymatgen/io/vasp/inputs.py

```diff
@@ -419,16 +420,43 @@ def from_str(cls, data, default_names=None, read_velocities=True):
                 for i, nat in enumerate(n_atoms):
                     sym = Element.from_Z(i + 1).symbol
                     atomic_symbols.extend([sym] * nat)
-                warnings.warn(f"Elements in POSCAR cannot be determined. Defaulting to false names {atomic_symbols}.")
-        # read the atomic coordinates
+                warnings.warn(
+                    f"Elements in POSCAR cannot be determined. Defaulting to false names {atomic_symbols}.",
+                    BadPoscarWarning,
+                )
+
+        # Read the atomic coordinates
         coords = []
         selective_dynamics = [] if has_selective_dynamics else None
         for i in range(n_sites):
             tokens = lines[ipos + 1 + i].split()
             crd_scale = scale if cart else 1
             coords.append([float(j) * crd_scale for j in tokens[:3]])
             if has_selective_dynamics:
-                selective_dynamics.append([tok.upper()[0] == "T" for tok in tokens[3:6]])
+                # Warn when values contain suspicious entries
+                if any(value not in {"T", "F"} for value in tokens[3:6]):
+                    warnings.warn(
+                        "Selective dynamics values must be either 'T' or 'F'.",
+                        BadPoscarWarning
+                        )
+
+                # Warn when elements contains Fluorine (F) (#3539)
+                if atomic_symbols[i] == "F" and len(tokens[3:]) >= 4 \
+                        and "F" in tokens[3:7]:
+                    warnings.warn(
+                        "Selective dynamics toggled with Fluorine element detected. Make sure the 4th-6th entry each position line is selective dynamics info.",
+                        BadPoscarWarning,
+                    )
+
+                selective_dynamics.append([value == "T" for value in tokens[3:6]])
+
+        # Warn when ALL degrees of freedom relaxed (#3539)
+        if has_selective_dynamics and all(all(i is True for i in in_list) for in_list in selective_dynamics):
+            warnings.warn(
+                "Ignoring selective dynamics tag, as no ionic degrees of freedom were fixed.",
```

Comment:
Is it looking better? @esoteric-ephemera Thanks!

Reply:
Perfect - both warnings should be clear to the user

Reply:
Thanks for reviewing @esoteric-ephemera